### PR TITLE
Remove provideRails and provideDemo flavour

### PR DESCRIPTION
**Pull request by andela-wakinsanmi**
We need to update the google-service.json package name to com.layer.messenger

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -36,8 +26,7 @@ dependencies {
 
     compile "com.google.firebase:firebase-messaging:10.2.0"
 
-    // Flavor-specific dependencies
-    providerdemoCompile 'com.google.android.gms:play-services-vision:10.2.0'
+    compile "com.google.android.gms:play-services-vision:10.2.0"
```
Comment (smpete) [inline app/build.gradle]: Do we need this anymore since the QR scanner is gone?

Comment (andela-wakinsanmi) [inline app/build.gradle]: No, we don't need it. I will get rid of it and update the branch

File: app/src/main/java/com/layer/messenger/flavor/util/CustomEndpoint.java [hunk 2]
```diff
@@ -151,6 +151,7 @@ public static Endpoint getEndpoint() {
                 return sEndpoints;
             } catch (JSONException e) {
                 if (Log.isLoggable(Log.ERROR)) Log.e(e.getMessage(), e);
+                throw new IllegalStateException(e);
```
Comment (smpete) [inline app/src/main/java/com/layer/messenger/flavor/util/CustomEndpoint.java]: Add a message to the exception using the (message, cause) constructor.

File: app/src/main/java/com/layer/messenger/App.java [hunk 3]
```diff
@@ -142,12 +136,10 @@ public void onDeauthenticationFailed(LayerClient client, String reason) {
     //==============================================================================================
 
     /**
-     * Gets or creates a LayerClient, using a default set of LayerClient.Options and flavor-specific
+     * Gets or creates a LayerClient, using a default set of LayerClient.Options
      * App ID and Options from the `generateLayerClient` method.  Returns `null` if the flavor was
```
Comment (smpete) [inline app/src/main/java/com/layer/messenger/App.java]: There's still a reference to "flavor" in this javadoc. That should be removed

File: app/src/main/java/com/layer/messenger/App.java [hunk 4]
```diff
@@ -164,9 +156,9 @@ public static LayerClient getLayerClient() {
                             ThreePartImageUtils.MIME_TYPE_PREVIEW));
 
             // Allow flavor to specify Layer App ID and customize Options.
-            sLayerClient = sFlavor.generateLayerClient(sInstance, options);
+            sLayerClient = generateLayerClient(sInstance, options);
 
-            // Flavor was unable to generate Layer Client (no App ID, etc.)
+            // unable to generate Layer Client (no App ID, etc.)
```
Comment (smpete) [inline app/src/main/java/com/layer/messenger/App.java]: Capitalize "unable" to maintain formatting consistency

Comment (smpete) [inline app/src/main/java/com/layer/messenger/App.java]: I think this was addressed wrong in line 141.

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -78,4 +78,11 @@
     <string name="notifications_new_messages">%1$d new messages</string>
 
     <string name="push_notification_no_content_title">New message (tap to sync)</string>
+
+    <string name="app_name">Atlas Messenger</string>
+    <string name="app_id_required">Please set your Layer App ID in `Flavor.java` when using the Rails Identity Provider.  You can get your Layer App ID from the developer dashboard.</string>
```
Comment (thecombatwombat) [inline app/src/main/res/values/strings.xml]: There is no `Flavor.java` any more. Please make sure this points to the correct location.

File: app/src/main/java/com/layer/messenger/RailsLoginActivity.java [hunk 6]
```diff
@@ -13,10 +13,8 @@
...
 import android.widget.Toast;
 
-import com.layer.messenger.App;
-import com.layer.messenger.ConversationsListActivity;
-import com.layer.messenger.R;
-import com.layer.messenger.flavor.util.CustomEndpoint;
+import com.layer.messenger.util.RailsAuthenticationProvider;
+import com.layer.messenger.util.CustomEndpoint;
 import com.layer.messenger.util.AuthenticationProvider;
 import com.layer.messenger.util.Log;
```
Comment (thecombatwombat) [inline app/src/main/java/com/layer/messenger/RailsLoginActivity.java]: Now that there is only one authentication provider and only one login activity, remove the `Rails` prefix from them.

File: app/src/main/java/com/layer/messenger/util/RailsAuthenticationProvider.java [hunk 7]
```diff
@@ -7,10 +7,8 @@
 import android.widget.Toast;
 
 import com.layer.messenger.R;
+import com.layer.messenger.RailsLoginActivity;
```
Comment (thecombatwombat) [inline app/src/main/java/com/layer/messenger/util/RailsAuthenticationProvider.java]: Now that there is only one authentication provider and only one login activity, remove the `Rails` prefix from them.

File: app/src/main/java/com/layer/messenger/RailsLoginActivity.java (hunk 6 above)
Comment (smpete) [inline app/src/main/java/com/layer/messenger/RailsLoginActivity.java]: That will change a lot of code. I'd suggest that name change in a separate PR

File: app/src/main/java/com/layer/messenger/util/RailsAuthenticationProvider.java (hunk 7 above)
Comment (smpete) [inline app/src/main/java/com/layer/messenger/util/RailsAuthenticationProvider.java]: That will change a lot of code. I'd suggest that name change in a separate PR

File: app/src/main/java/com/layer/messenger/RailsLoginActivity.java (hunk 6 above)
Comment (thecombatwombat) [inline app/src/main/java/com/layer/messenger/RailsLoginActivity.java]: @andela-wakinsanmi ok... Lets merge these, but please follow up with a PR that removes the `Rails` prefixes on these

**Pull request merged**